lib/model/SignUpStatus.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:openfoodfacts/openfoodfacts.dart';
+
+/// The response from a call to the [OpenFoodAPIClient.register] method
+/// The [status] may be:
+/// - 201: Account created + user logged in
+/// - 400: An error occurred - see [statusError] for more information
+/// - 500: Unknown error
+class SignUpStatus extends Status {
+  final SignUpStatusError? statusError;
+
+  SignUpStatus._({
+    dynamic status,
+    String? statusVerbose,
+    String? body,
+    String? error,
+    int? imageId,
+    this.statusError,
+  }) : super(
+          status: status,
+          statusVerbose: statusVerbose,
+          body: body,
+          error: error,
+          imageId: imageId,
+        );
+
+  factory SignUpStatus(Status status) {
+    if (status.body == null) {
+      return SignUpStatus._(
+        status: 500,
+        statusError: SignUpStatusError.UNKNOWN,
+        error:
+            'No response, open an issue here: https://github.com/openfoodfacts/openfoodfacts-dart/issues/new',
+      );
+    } else if (status.body!.contains('loggedin')) {
+      return SignUpStatus._(
+        status: 201,
+      );
+    } else {
+      return SignUpStatus._(
+        status: 400,
+        statusError: _findErrorInMap(status.body) ?? SignUpStatusError.UNKNOWN,
+        error: _extractErrorFromHTMLBody(status.body),
+      );
+    }
+  }
+
+  /// [_errorTexts] have values where have to call the [contains] method
+  static _findErrorInMap(String? sentence) {
+    if (sentence == null) {
+      return null;
+    }
+
+    for (String errorLabel in _errorTexts.keys) {
+      if (sentence.contains(errorLabel)) {
+        return _errorTexts[errorLabel];
+      }
+    }
+
+    return null;
+  }
+
+  /// Try to extract the error from the HTML… or will return the [body] otherwise
+  static String? _extractErrorFromHTMLBody(String? body) {
```

Comment:
Now that I'm thinking we would probably be better off just `return`ing the body: same behavior as before, no change.
Not very important though.